sdk/python/feast/infra/materialization/snowflake/snowflake_engine.py

```diff
@@ -0,0 +1,243 @@
+        self, registry, tasks: List[MaterializationTask]
+    ) -> List[MaterializationJob]:
+        return [
+            self._materialize_one(
+                registry,
+                task.feature_view,
+                task.start_time,
+                task.end_time,
+                task.project,
+                task.tqdm_builder,
+            )
+            for task in tasks
+        ]
+
+    def _materialize_one(
+        self,
+        registry: BaseRegistry,
+        feature_view: Union[BatchFeatureView, StreamFeatureView, FeatureView],
+        start_date: datetime,
+        end_date: datetime,
+        project: str,
+        tqdm_builder: Callable[[int], tqdm],
+    ):
+        assert isinstance(feature_view, BatchFeatureView) or isinstance(
+            feature_view, FeatureView
+        ), "Snowflake can only materialize batch features views."
+
+        entities = []
+        for entity_name in feature_view.entities:
+            entities.append(registry.get_entity(entity_name, project))
+
+        (
+            join_key_columns,
+            feature_name_columns,
+            timestamp_field,
+            created_timestamp_column,
+        ) = _get_column_names(feature_view, entities)
+
+        job_id = f"{feature_view.name}-{start_date}-{end_date}"
+
+        try:
+            offline_job = self.offline_store.pull_latest_from_table_or_query(
+                config=self.repo_config,
+                data_source=feature_view.batch_source,
+                join_key_columns=join_key_columns,
+                feature_name_columns=feature_name_columns,
+                timestamp_field=timestamp_field,
+                created_timestamp_column=created_timestamp_column,
+                start_date=start_date,
+                end_date=end_date,
+            )
+
+            table = offline_job.to_arrow()
+            fv_latest_values_sql = offline_job.to_sql()
+
+            """
+            if feature_view.batch_source.field_mapping is not None:
+                fv_latest_values_sql = _run_pyarrow_field_mapping(
+                    table, feature_view.batch_source.field_mapping
+                )
+            """
+
+            join_key_to_value_type = {
+                entity.name: entity.dtype.to_value_type()
+                for entity in feature_view.entity_columns
+            }
+
+            fv_columns = [
+                (field.name, field.dtype.to_value_type())
+                for field in feature_view.features
+            ]
+
+            fv_timestamp = feature_view.batch_source.timestamp_field
+
+            if feature_view.batch_source.created_timestamp_column:
+                fv_created = feature_view.batch_source.created_timestamp_column
+            else:
+                fv_created = None
+
+            online_table = f"""{self.repo_config.online_store.database}"."{self.repo_config.online_store.schema_}"."{self.repo_config.project}_{feature_view.name}"""
+
+            with Session.builder.configs(
```

Comment:
@achals @adchia There actually might be a code path that doesnt need snowpark, I will just need to take these inputs and build a dynamic sql query -- which might be the most efficient anyways